app/src/main/java/ch/epfl/sdp/blindly/match/Profile.kt

```diff
@@ -32,7 +36,13 @@ data class Profile(
     /**
      * A constructor for a profile from a parcel
      */
-    constructor(source: Parcel) : this(source.readString()!!, source.readInt())
+    constructor(source: Parcel) : this(
+        source.readString()!!,
+        source.readInt(),
```

Comment:
Sorry if this is basic but at this point I'm a bit distant to this part of the code. Why the second element of the constructor doesn't have a non-null assertion operator? In any way, !!s might cause us some problems afterwards but maybe that was the only way.

Reply:
It's because readString returns a String? but readInt returns an Int. I guess we could store the variables as String?s but we would have to handle the uncertainty all the way through the activities. I think it's easier to assume the check is done before constructing the profile